src/AppEntry.js

```diff
@@ -1,29 +1,26 @@
-import React, { useState, useEffect } from 'react';
+import React, { useMemo } from 'react';
 import { BrowserRouter as Router } from 'react-router-dom';
 import { Provider } from 'react-redux';
-import { init, RegistryContext } from './store';
+import { getStore, RegistryContext, updateReducers } from './store';
 import App from './App';
 import { getBaseName } from '@redhat-cloud-services/frontend-components-utilities/helpers';
 import logger from 'redux-logger';
 import Fallback from './components/SpinnerFallback';
 
 const InventoryApp = () => {
-    const [registry, setRegistry] = useState();
-    const store = registry?.getStore();
-
-    useEffect(() => {
-        setRegistry(IS_DEV ? init(logger) : init());
-
-        return () => {
-            setRegistry(undefined);
+    const registry = useMemo(() => {
```

Comment:
Why remove using the state? Memo as far as i understand it can be thrown away any time React likes to, or?

Reply:
The difference here is that `useMemo` doesn't generate new re-render. However using `useEffect` - `useState` will cause new render right after the initial mount.

Reply:
I like lesser re-render. I do not see any issues with useEffect which has an empty dependency array. I did some googling.

Ref: https://beta.reactjs.org/reference/react/useMemo#my-calculation-runs-twice-on-every-re-render

Reply:
I like this approach let's fix the tests and merge if everything works as expected. I want to add some of my changes as well :)

Reply:
But we want it to re-render, do we not? It is also a bit iffy that the registry in the memo is not under our, but React's control.

Reply:
Considering your concern about useMemo's unwanted recalculating behavior, I do not see anything that can lead to it in the react docs or anywhere [React caveats](https://beta.reactjs.org/reference/react/useMemo#caveats). Maybe, I am well wrong and happy to learn.

Reply:
@bastilian the other way around actually. We don't want it to re-render, in the initial render we calculate the value of provider and don't need additional renders.

Reply:
@karelhala "re-render" as in ensure that a value is set. In this case the advice the advice from the React docs to only use `useMemo` as a performance optimisation holds up here as well, as without using the useMemo the app would not work properly and has a chance of fail. This is why I feel a`useState` is more appropriate here and elsewhere where we need to have a registry for child components.

Reply:
@bastilian both `useEffect` with `useState` has same outcome as `useMemo`. We have to create the redux store only on initial render. This can be achieved either by using `useEffect` and setting the value in `useState, however this causes new render which can be optimized using `useMemo`. Basically `useEffect` is good for side effects, like querying server, dispatching new actions to redux. Whereas `useMemo` is used to memoize the value between renderes, meaning we run this code once on initial render, rember the value and not trigger new render. This approach has been used for instance in advisor https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/AppEntry.js#L13 because previous way (setting on each render) was causing re-creating redux store.

Some examples: https://medium.com/@zoenkatz/usememo-vs-usestate-and-useeffect-d906901d88c https://angeloocana.com/en/blog/react/why-usememo-instead-of-usestate-useeffect/

Reply:
@karelhala The thing here is that the instance of the registry/store has to stay exactly the same, which I don't trust `useMemo` can guarantee.

I'm fine with using the `useMemo` here, but I do not trust it and if there are any issues related to this, we should revert this to be in a state where the initialisation is only controlled by us.

Reply:
How would useState make sure to have the same instance across page re-visits? I would think it would create a new instance from scratch.

Having `useState` and React context together to have the same instance always produced the same issue, as far as I understand.

Reply:
@bastilian @karelhala merging this so that we can test it on stage env. In case there is any issue, it is not difficult to change `useMemo` to `useState`. If this is totally wrong, please ping me and I will revert.